adapters/tlsio_appleios.c

```diff
@@ -0,0 +1,546 @@
+int tlsio_appleios_close(CONCRETE_IO_HANDLE tls_io, ON_IO_CLOSE_COMPLETE on_io_close_complete, void* callback_context);
+int tlsio_appleios_send(CONCRETE_IO_HANDLE tls_io, const void* buffer, size_t size, ON_SEND_COMPLETE on_send_complete, void* callback_context);
+void tlsio_appleios_dowork(CONCRETE_IO_HANDLE tls_io);
+int tlsio_appleios_setoption(CONCRETE_IO_HANDLE tls_io, const char* optionName, const void* value);
+OPTIONHANDLER_HANDLE tlsio_appleios_retrieveoptions(CONCRETE_IO_HANDLE tls_io);
+
+static const IO_INTERFACE_DESCRIPTION tlsio_handle_interface_description =
+{
+    tlsio_appleios_retrieveoptions,
+    tlsio_appleios_create,
+    tlsio_appleios_destroy,
+    tlsio_appleios_open,
+    tlsio_appleios_close,
+    tlsio_appleios_send,
+    tlsio_appleios_dowork,
+    tlsio_appleios_setoption
+};
+
+/* Codes_SRS_TLSIO_APPLEIOS_32_027: [ The tlsio_appleios_open shall set the tlsio to try to open the connection for 10 times before assuming that connection failed. ]*/
+#define MAX_TLS_OPENING_RETRY  10
+/* Codes_SRS_TLSIO_APPLEIOS_32_044: [ The tlsio_appleios_close shall set the tlsio to try to close the connection for 10 times before assuming that close connection failed. ]*/
+#define MAX_TLS_CLOSING_RETRY  10
+#define RECEIVE_BUFFER_SIZE    128
+
+#define CallErrorCallback() do { if (tlsio_instance->on_io_error != NULL) (void)tlsio_instance->on_io_error(tlsio_instance->on_io_error_context); } while((void)0,0)
+#define CallOpenCallback(status) do { if (tlsio_instance->on_io_open_complete != NULL) (void)tlsio_instance->on_io_open_complete(tlsio_instance->on_io_open_complete_context, status); } while((void)0,0)
+#define CallCloseCallback() do { if (tlsio_instance->on_io_close_complete != NULL) (void)tlsio_instance->on_io_close_complete(tlsio_instance->on_io_close_complete_context); } while((void)0,0)
+
+
+typedef struct AppleiosTLS_tag
+{
+    ON_IO_OPEN_COMPLETE on_io_open_complete;
+    void* on_io_open_complete_context;
+
+    ON_BYTES_RECEIVED on_bytes_received;
+    void* on_bytes_received_context;
+
+    ON_IO_ERROR on_io_error;
+    void* on_io_error_context;
+
+    ON_IO_CLOSE_COMPLETE on_io_close_complete;
+    void* on_io_close_complete_context;
+
+    CFStringRef remote_host;
+    uint16_t port;
+	
+    CFReadStreamRef sockRead;
+    CFWriteStreamRef sockWrite;
+	
+    TLSIO_APPLEIOS_STATE state;
+    int countTry;
+} AppleiosTLS;
+
+/* Codes_SRS_TLSIO_APPLEIOS_32_008: [ The tlsio_appleios_get_interface_description shall return the VTable IO_INTERFACE_DESCRIPTION. ]*/
+const IO_INTERFACE_DESCRIPTION* tlsio_appleios_get_interface_description(void)
+{
+    return &tlsio_handle_interface_description;
+}
+
+
+/* Codes_SRS_TLSIO_APPLEIOS_32_020: [ If tlsio_appleios_create get success to create the tlsio instance, it shall set the tlsio state as TLSIO_APPLEIOS_STATE_CLOSED. ]*/
+TLSIO_APPLEIOS_STATE tlsio_appleios_get_state(CONCRETE_IO_HANDLE tlsio_handle)
```

Comment:
>tlsio_appleios_get_state [](start = 21, length = 24)

This should be removed as the state should anyhow be removed from the public API.